lms/services/file.py

```diff
@@ -57,32 +64,30 @@ def upsert(self, file_dicts):
         )
 
     def _file_search_query(
-        self,
-        application_instance,
-        type_,
-        *,
-        lms_id=None,
-        course_id=None,
-        name=None,
-        size=None
+        self, guid, type_, *, lms_id=None, course_id=None, name=None, size=None
     ):
         """Return a `File` query with the passed parameters applied as filters."""
-        query = self._db.query(File).filter_by(
-            application_instance_id=application_instance.id,
-            type=type_,
+        query = (
+            self._db.query(File)
+            .join(ApplicationInstance)
+            .filter(
+                # We don't query by application_instance but any file belonging to an AI with the same GUID
+                ApplicationInstance.tool_consumer_instance_guid == guid,
```

Comment:
We should bring this in a BE discussion

Is storing by application instance but querying by GUID a good model to apply in general?

I reckon is the best option without a big refactor (almost rewrite of some parts), the alternatives are:

- Store by GUID directly. This would save a few lines while querying but we'd lose some information (the application instances changes / relationships) and adapting existing tables would required complex migrations.

- Store by organization. This is probably the direction we should move eventually. Right now the organization data is the DB is not yet complete (ie all installs of the same school might not belong to the same org yet).

Even if we stored things by organization we'd still need to use the GUID to communicate with H, generate group and user IDs.

-  Maybe a middle ground, store by GUID but have another `model_application_instance` table to keep record of the applications instances we seen related to that model. 
 
In this case we'd have `file_application_instance`. Again it will need complex migration for existing models but probably a more achievable goal than the organization one.